server/service/src/programs/encounter/update.rs

```diff
@@ -51,16 +51,38 @@ pub fn update_encounter(
             let encounter_start_datetime = encounter.start_datetime;
             let doc = generate(user_id, input, existing)?;
 
-            if is_latest_doc(ctx, service_provider, &doc)
+            let document = service_provider
```

Comment:
does this relocation mean there is validation first before encounter row update ?

Reply:
there was no reason for the re-position of the code. The reason simply that I didn't had the right data type when doing the sync where I had a `Document` and before this change it was working on with a `RawDocument`... So now I write the document and if it is the latest doc I also update the encounter table which seems also be the more nature flow.